src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -318,10 +320,13 @@ public void saveAs(File file) throws Exception {
 
         if (context.getLocation() == DatabaseLocation.SHARED) {
             // Save all properties dependent on the ID. This makes it possible to restore them.
-            DBMSConnectionProperties properties = context.getDBMSSynchronizer().getDBProcessor()
-                    .getDBMSConnectionProperties();
-            new SharedDatabasePreferences(context.getDatabase().generateSharedDatabaseID())
-                    .putAllDBMSConnectionProperties(properties);
+            DatabaseSynchronizer synchronizer = context.getDBMSSynchronizer();
+            if (synchronizer instanceof DBMSSynchronizer) {
+                DatabaseConnectionProperties properties = ((DBMSSynchronizer) synchronizer).getDBProcessor()
```

Comment:
If you add `getDBMSConnectionProperties` to the `Synchronizer` interface, then you probably don't need the if statement and the explicit cast here.

Reply:
Thanks, nicely spotted! So far, I was focused on the refactoring only, but the code certainly has a lot of potential for improvements. Implementing your suggestion was fairly easy and I fixed this at a second position as well.